File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program105.java
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and count number of white spaces
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Solution
{
	String str;
	int iCnt,iCount;

	Solution(String str)
	{
		this.str=str;
		this.iCnt=0;
		this.iCount=0;
	}

	int CountWhite()
	{
		
		while(iCnt!=str.length())
		{
			if(str.charAt(iCnt)==' ')
			{
				iCount++;
			}
			iCnt++;
		}
		
		return iCount;
	}
}

class Program
{
	public static void main(String args[])
	{
		int iRet=0;
		Scanner sObj = new Scanner(System.in);
		System.out.println("Enter the string: ");
		String str = sObj.nextLine();

		Solution obj = new Solution(str);
		iRet = obj.CountWhite();

		System.out.println("White spaces are: "+iRet);
	}
}

/*
OUTPUT:

Enter the string:
Vishal Tejkant Patil
White spaces are: 3

*/

File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program105.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and count number of white spaces
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{

	private:
		string str;
		int iCnt,iCount;

	public:

		Solution(string str,int iCnt=0,int iCount=0)
		{
			this->str=str;
		}

		int CountWhite()
		{

			while(iCnt!=str.length())
			{
				if(str[iCnt]==' ')
				{
					iCount++;
				}

				iCnt++;
				
			}	
			return iCount;
		}
};

int main()
{
	string str;
	int iRet=0;

	cout<<"Enter string: ";
	getline(cin,str);
	
	Solution *sObj = new Solution(str);
	iRet=sObj->CountWhite();

	cout<<"White spaces are: "<<iRet<<"\n";
	delete sObj;
	return 0;

}

/*
OUTPUT:

Enter string: MarvellouS Infosystems by Piysuh Manohar Khairnar
White spaces are: 5

*/
